sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -2080,8 +2080,10 @@ case class ParseToTimestamp(
   override def inputTypes: Seq[AbstractDataType] = {
     // Note: ideally this function should only take string input, but we allow more types here to
     // be backward compatible.
-    TypeCollection(StringType, DateType, TimestampType, TimestampNTZType) +:
-      format.map(_ => StringType).toSeq
+    val types = Seq(StringType, DateType, TimestampType, TimestampNTZType)
```

Comment:
BTW, it's from `Cast` logic - we used `Cast` here before https://github.com/apache/spark/commit/fab4ceb157baac870f6d50b942084bb9b2cd4ad2:
https://github.com/apache/spark/blob/c1e5e8a27595c711c53bb3571ea391ef700d531a/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala#L71